### PR TITLE
Shortcuts: Require modifier key to trigger shortcuts (`F`,`A`,`D`,`S`,`T`,`/`)

### LINKED ISSUE #16022: Consider changing S / D / F / T / A keyboard shortcuts
**Is your feature request related to a problem? Please describe**
These shortcuts seem to be offputting and unexpected to many, especially because they're easy to accidentally trigger. 

**Describe the solution you'd like**
We should consider requiring a modifier key for these shortcuts.

**Describe alternatives you've considered**
Drop them altogether. None of them are really used frequently enough to warrant a shortcut.

**Are you able to assist to bring the feature to reality?**
Totally.

### PR DESCRIPTION
**Pull request by cdedreuille**
Closes #16022

## What I did

In this PR I'm modifying the default shortcuts to avoid having shortcuts being triggered unintentionally. I added `Alt` before the first 6 shortcuts transforming from `S` to `Alt + S`.

I also replaced `/` for search into `command + K` as we are exploring updating our search into a command palette and `command + k` is a well known shortcut for that.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

File: code/lib/manager-api/src/modules/shortcuts.ts [hunk 1]
```diff
@@ -125,12 +125,12 @@ type API_AddonShortcutLabels = Record<string, string>;
...
-  panelPosition: ['D'],
-  toggleNav: ['S'],
-  toolbar: ['T'],
-  search: ['/'],
+  fullScreen: [controlOrMetaKey(), 'F'],
+  togglePanel: [controlOrMetaKey(), 'A'],
+  panelPosition: [controlOrMetaKey(), 'D'],
+  toggleNav: [controlOrMetaKey(), 'S'],
+  toolbar: [controlOrMetaKey(), 'T'],
+  search: [controlOrMetaKey(), 'K'],
```
[the remark this answers is not in the document]
Comment (cdedreuille) [reply code/lib/manager-api/src/modules/shortcuts.ts:133]: Thanks @ndelangen I updated them now.

**Review by JReinhold — CHANGES_REQUESTED** [review]
Code looks good, but it needs a migration note.

**Comment by cdedreuille** [comment]
@JReinhold I updated the shortcuts to use `Alt` instead of `command for most of them except for search that will stay as `command + K` to match what others are doing with their command action. I also added migration guides as you suggested.

**Comment by cdedreuille** [comment]
It seems we have flaky E2E tests. Since all other tests passed, we discussed this with @JReinhold and he gave me approval to merge this PR directly since the E2E bug is very unlikely related to this issue.

**Comment by adrian-cocoara-lgp** [comment]
Hello @cdedreuille,
These changes broke some of our Test Cases since the shortcuts keys have changed.
I checked the documentation if there is any way these shortcuts are configurable, but I can't find anything. 
Is there any way I can revert them to the values of Storybook 7? Thanks!

**Pull request merged**